/******************************************************************************
 *
 *  Copyright (c) 2014-2015, 2020-2021 The Linux Foundation. All rights reserved
 *  Copyright (C) 2003-2012 Broadcom Corporation
 *  Copyright (C) 2008, 2011-2012, 2016-2018 The Android Open Source Project
 *  Copyright (C) 2014 Google, Inc
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at:
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 ******************************************************************************/

From b513db3e630666ff4b58fc3a5f568aad4d9214cd Mon Sep 17 00:00:00 2001
From: Sunny Kapdi <sunnyk@codeaurora.org>
Date: Sat, 13 Nov 2021 18:23:19 -0800
Subject: [PATCH] BT5.3: Add support for ADI in Periodic Advertising

Add support for ADI in Periodic Advertising Enable command

Change-Id: Ifc90f65f45a9f0973a16ced1286bfe064de509de
---
 jni/com_android_bluetooth_gatt.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jni/com_android_bluetooth_gatt.cpp b/jni/com_android_bluetooth_gatt.cpp
index 949abdbac..0aa5a2d60 100644
--- a/jni/com_android_bluetooth_gatt.cpp
+++ b/jni/com_android_bluetooth_gatt.cpp
@@ -2095,7 +2095,7 @@ static PeriodicAdvertisingParameters parsePeriodicParams(JNIEnv* env,
   methodId = env->GetMethodID(clazz, "getInterval", "()I");
   uint16_t interval = env->CallIntMethod(i, methodId);
 
-  p.enable = true;
+  p.enable = 0x03; // Set bit0 (enable) and bit1 (ADI)
   p.min_interval = interval;
   p.max_interval = interval + 16; /* 20ms difference betwen min and max */
   uint16_t props = 0;
-- 
2.25.1

